Replace usages of impl.factory.(Bags|Lists|Sets|Maps) with api.factory in unit-tests.

motlin opened the pull request.

donraab reviewed (approved):
LGTM. It would be great if you could update all of the copyright years in these files as well.

motlin commented:
> LGTM. It would be great if you could update all of the copyright years in these files as well.

Updated Copyrights in a second commit, and rebased both commits.

donraab reviewed (approved):
LGTM
